docs/providers/aws/guide/iam.md

```diff
@@ -0,0 +1,279 @@
+
+If you want to give permission to your functions to access certain resources on your AWS account, you can add custom IAM role statements to your service by adding the statements in the `iamRoleStatements` array in the `provider` object. As those statements will be merged into the CloudFormation template you can use `Join`, `Ref` or any other CloudFormation method or feature. You're also able to either use YAML for defining the statement (including the methods) or use embedded JSON if you prefer it. Here's an example that uses all of these:
+
+```yml
+service: new-service
+
+provider:
+  name: aws
+  iamRoleStatements:
+    -  Effect: "Allow"
+       Action:
+         - "s3:ListBucket"
+       Resource: { "Fn::Join" : ["", ["arn:aws:s3:::", { "Ref" : "ServerlessDeploymentBucket"} ] ] }
+    -  Effect: "Allow"
+       Action:
+         - "s3:PutObject"
+       Resource:
+         Fn::Join:
+           - ""
+           - - "arn:aws:s3:::"
+             - "Ref" : "ServerlessDeploymentBucket"
+```
+
+On deployment, all these statements will be added to the policy that is applied to the IAM role that is assumed by your Lambda functions.
+
+## Custom Role Management
+
+Serverless empowers you to define custom roles and apply them to your functions on a provider or individual function basis. To do this you must declare a `role` attribute at the level at which you would like the role to be applied.
+
+Defining it on the provider will make the role referenced by the `role` value the default role for any Lambda without its own `role` declared. This is to say that defining a `role` attribute on individual functions will override any provider level declared role. If every function within your service has a role assigned to it (either via provider level `role` declaration, individual declarations, or a mix of the two) then the default role and policy will not be generated and added to your Cloud Formation Template.
+
+The `role` attribute can have a value of the logical name of the role, the ARN of the role, or an object that will resolve in the ARN of the role. The declaration `{ function: { role: 'myRole' } }` will result in `{ 'Fn::GetAtt': ['myRole', 'Arn'] }`. You can of course just declare an ARN like so `{ function: { role: 'an:aws:arn:xxx:*:*' } }`. This use case is primarily for those who must create their roles and / or policies via a means outside of Serverless.
+
+Here are some examples of using these capabilities to specify Lambda roles.
+
+### Provide a single role for all lambdas (via each form of declaration)
+
+```yml
+service: new-service
+
+provider:
+  name: aws
+  # declare one of the following...
+  role: myDefaultRole                                                # must validly reference a role defined in the service
+  role: arn:aws:iam::0123456789:role//my/default/path/myDefaultRole  # must validly reference a role defined in your account
```

Comment:
if you're referencing roles created outside of Serverless, it's best if we change the `myDefaultRole` name because  it gives the impression that you're referencing the custom resources defined below